praw/reddit.py

```diff
@@ -579,10 +634,30 @@ def post(
             None).
 
         """
-        data = self.request(
-            "POST", path, data=data or {}, files=files, params=params
-        )
-        return self._objector.objectify(data)
+        data = data or {}
+        try:
+            return self._objectify_request(
+                data=data,
+                files=files,
+                method="POST",
+                params=params,
+                path=path,
+            )
+        except RedditAPIException as exception:
+            seconds = self._handle_rate_limit(exception=exception)
+            if seconds is not None:
+                logger.debug(
+                    "Rate limit hit, sleeping for %.2f seconds", seconds,
+                )
+                time.sleep(seconds)
+                return self._objectify_request(
+                    data=data,
+                    files=files,
+                    method="POST",
+                    params=params,
+                    path=path,
+                )
+            raise exception
```

Comment:
This can just be `raise` to re-raise the exception with the original stacktrace.